webhooks/pod/mutate_image_path.go

```diff
@@ -69,25 +64,39 @@ func (ipr *ImagePathRewriter) Handle(ctx context.Context, req admission.Request)
 		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("get pod namespace object error: %w", err))
 	}
 
-	ipr.Log.Info("receive pod request", "pod", pod)
+	ipr.Log.V(4).Info("receive pod request", "pod", pod)
+
+	flag, flagPresent := podNS.Annotations[annotationRewriter]
+	if flag == imageRewriteDisabled {
+		return admission.Allowed("image rewriting disabled")
+	}
+	// If the rewrite annotation is not set, fallback to default hsc handling
+	if !flagPresent {
+		defaultHSC, err := ipr.lookupDefaultHarborServerConfig(ctx)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, fmt.Errorf("get default hsc object error: %w", err))
+		}
+		if defaultHSC == nil {
+			return admission.Allowed("no default harbor server configuration")
+		}
 
+		return ipr.rewriteContainers(req, defaultHSC.Spec.ServerURL, defaultHSC.Spec.Rules, pod)
```

Comment:
The transparent proxy feature is supported here, right? A rewrite rule like 

```go
[{
   registry: "docker.io",
   harborProject: "harbor_proxy_cache_project",
},{
   ...
}]
```
?